tests/sentry/sentry_metrics/limiters/test_cardinality_limiter.py

```diff
@@ -65,125 +66,119 @@ def use_quotas(
 
 
 def test_reject_all(set_sentry_option):
-    set_sentry_option(
+    with set_sentry_option(
```

Comment:
In a case like this where the arguments to `set_sentry_option` do not depend on any function parameters, could `set_sentry_option` also be a decorator?

Reply:
Definitely.  But if I'm honest, I'm inclined to leave that without making deeper changes for the time being.  Given that the current system causes subtle test bugs, I want one known "good" commit to separate HC changes from this context test bug.